lib/substitution_map_extractor.ex

```diff
@@ -7,26 +7,43 @@ defmodule SubstitutionMapExtractor do
   @order_number_regex ~r|^#(\d+)$|
 
   # Massages a single file
-  def extract_substitution_map(file_path) do
-    file_path
+  def extract_substitution_map(%{
+        file_name: file_name,
+        previous_file_last_order_number: previous_file_last_order_number
+      }) do
+    file_name
     |> Path.expand()
     |> Path.absname()
     |> IO.inspect(label: "Extracting substitution map from:")
     |> File.stream!()
     |> CSV.decode!()
-    |> convert()
+    |> convert(previous_file_last_order_number)
   end
 
-  def convert(decoded_csv) do
-    decoded_csv
-    # Remove the first line, it's just headers
-    |> Enum.drop(1)
-    |> convert_to_order_number_name_map()
-    |> convert_mapped()
-    |> Enum.filter(fn %{is_valid: is_valid} = _row_map -> not is_valid end)
-    |> Enum.map(fn %{order_number: order_number} = conversion_map ->
-      conversion_map |> Map.put(:replace_order_number, "##{order_number}")
-    end)
+  def convert(decoded_csv, previous_file_last_order_number) do
+    raw_mapped =
+      decoded_csv
+      # Remove the first line, it's just headers
+      |> Enum.drop(1)
+      # We go from the last entry to the first entry
+      # which is equivalent to figuring out the substitutions from the oldest
+      # entry to the newest entry.
+      # We do this because the very first entry is likely #1001, and we can
+      # usually have accurate numbering from this point.
+      |> Enum.reverse()
+      |> convert_to_order_number_name_map()
+      |> convert_mapped(previous_file_last_order_number)
+
+    filtered =
+      raw_mapped
+      |> Enum.filter(fn %{is_valid: is_valid} = _row_map -> not is_valid end)
+      |> Enum.map(fn %{order_number: order_number} = conversion_map ->
+        conversion_map |> Map.put(:replace_order_number, "##{order_number}")
+      end)
+
+    [%{order_number: file_last_order_number} | _] = raw_mapped |> Enum.reverse()
```

Comment:
We need to reverse here so that the first entry in the list is the _last_ entry of the file (which seeds the previous order number for the next file).